api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -157,6 +157,11 @@ message TlsParameters {
   //   rsa_pkcs1_sha1
   //   ecdsa_sha1
   repeated string signature_algorithms = 5;
+
+  // By default, Envoy as a server uses its preferred cipher during the handshake.
+  // Setting this to true would allow the downstream client's preferred cipher to be used instead.
+  // Has no effect when using TLSv1_3.
+  bool prefer_client_ciphers = 6;
```

Comment:
Should this field be in `DownstreamTlsContext` instead of here?  If I'm understanding the intent here correctly, this seems to apply to the downstream connection only, not to upstream connections.  But the `TlsParameters` message is used in `CommonTlsContext`, which is used in both `UpstreamTlsContext` and `DownstreamTlsContext`.

Reply:
That's a good catch! Yes, agree this should be in the downstream context

Reply:
Oops, Thanks for pointing that out. I've moved it to downstream context now.